src/h5web/visualizations/nexus/utils.ts

```diff
@@ -124,15 +125,49 @@ export function getLinkedDatasets(
   return Object.fromEntries(datasetEntries);
 }
 
-export function getNxAxesParams(
-  nxDatasets: Record<string, HDF5Dataset>,
-  datasetValues: Record<string, HDF5Value> | undefined
-): Record<string, AxisParams> {
-  const paramsEntries = Object.entries(nxDatasets).map(([axis, dataset]) => {
-    const values = datasetValues && datasetValues[axis];
-    assertOptionalArray<number>(values);
-    return [axis, { values, label: getDatasetLabel(dataset, axis) }];
+export function useParsedNxData(
+  group: HDF5Group,
+  metadata: HDF5Metadata
+): NxData | undefined {
+  const datasetValues = useChildDatasetValues(group, metadata);
+  if (!datasetValues) {
+    return undefined;
+  }
+
+  const signalName = getAttributeValue(group, 'signal');
+  assertStr(signalName);
+  const signalDataset = getLinkedEntity(signalName, group, metadata);
+  const signalValue = datasetValues[signalName];
+
+  if (!signalDataset || !isDataset(signalDataset) || !signalValue) {
+    return undefined;
+  }
+  assertArray<number>(signalValue);
+
+  const signal = {
+    label: getDatasetLabel(signalDataset, signalName),
+    value: signalValue,
+    dims: (signalDataset.shape as HDF5SimpleShape).dims,
+  };
```

Comment:
I would just move this object as a nested object in the `return` statement and wouldn't bother with a `signal` constant.